packages/types/src/token.ts

```diff
@@ -1,18 +1,14 @@
+import { TokenIdentity } from "./identity";
 import { Provider } from "./util";
 
 /**
  * An object representing temporary or permanent AWS token.
+ *
+ * @deprecated Use {@TokenIdentity}
  */
-export interface Token {
-  /**
-   *The literal token string
-   */
-  readonly token: string;
-  /**
-   * A {Date} when these token will no longer be accepted.
-   * When expiration is not defined, the token is assumed to be permanent.
-   */
-  readonly expiration?: Date;
-}
+export interface Token extends TokenIdentity {}
 
+/**
+ * @deprecated Use {@TokenIdentityProvider}
+ */
 export type TokenProvider = Provider<Token>;
```

Comment:
Should this be the following?

```suggestion
/**
 * @deprecated Use {@TokenIdentityProvider}
 */
export type TokenProvider = IdentityProvider<Token>;
```